Core/include/Acts/Surfaces/Surface.hpp

```diff
@@ -434,35 +427,48 @@ class Surface : public virtual GeometryObject,
   /// extrinsic Euler angles)
   ///
   /// @param gctx The current geometry context object, e.g. alignment
-  /// @param derivatives Path length derivatives of the free, nominal
-  /// parameters to help evaluate change of free track parameters caused by
   /// change of alignment parameters
-  /// @param position The position of the paramters in global
-  /// @param direction The direction of the track
+  /// @param parameters is the free parameters
+  /// @param pathDerivative is the derivative of free parameters w.r.t. path
+  /// length
   ///
   /// @return Derivative of bound track parameters w.r.t. local frame
   /// alignment parameters
   AlignmentToBoundMatrix alignmentToBoundDerivative(
-      const GeometryContext& gctx, const FreeVector& derivatives,
-      const Vector3D& position, const Vector3D& direction) const;
+      const GeometryContext& gctx, const FreeVector& parameters,
+      const FreeVector& pathDerivative) const;
 
-  /// Calculate the derivative of path length w.r.t. alignment parameters of the
-  /// surface (i.e. local frame origin in global 3D Cartesian coordinates and
-  /// its rotation represented with extrinsic Euler angles)
+  /// Calculate the derivative of bound track parameters w.r.t.
+  /// alignment parameters of its reference surface (i.e. origin in global 3D
+  /// Cartesian coordinates and its rotation represented with extrinsic Euler
+  /// angles) without any path correction
   ///
-  /// Re-implementation is needed for surface whose intersection with track is
-  /// not its local xy plane, e.g. LineSurface, CylinderSurface and ConeSurface
+  /// @note This function should be used together with alignment to path
+  /// derivative to get the full alignment to bound derivatives
   ///
   /// @param gctx The current geometry context object, e.g. alignment
-  /// @param rotToLocalZAxis The derivative of local frame z axis vector w.r.t.
-  /// its rotation
-  /// @param position The position of the paramters in global
-  /// @param direction The direction of the track
+  /// @param parameters is the free parameters
+  ///
+  /// @return Derivative of bound track parameters w.r.t. local frame alignment
+  /// parameters without path correction
+  AlignmentToBoundMatrix alignmentToBoundDerivativeWithoutCorrection(
+      const GeometryContext& gctx, const FreeVector& parameters) const;
```

Comment:
Hi @msmk0 , thanks! 
Yes, this function is contained in the `alignmentToBound` originally. It's `private` now.

For the local-to-global Jacobian, we could open another PR for it since we might need to fix `freeToPathDerivative` for the `ConeSurface` and `CylinderSurface` additionally, and this might require some discussion.

Reply:
Yes, let's handle that in a separate PR. I approved this one and will merge it once the CI runs through.